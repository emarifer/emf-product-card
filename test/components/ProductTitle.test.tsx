import React from 'react';
import renderer from 'react-test-renderer';
import { ProductTitle, ProductCard } from '../../src/components';
import { product1 } from '../data/products';

describe('ProductTitle', () => {
  test('it should display the component correctly with the custom title', () => {
    const wrapper = renderer.create(
      <ProductTitle title="Custom Product - Title" className="custom3-class" />
    );
    // console.log(wrapper.toJSON());
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('it should show the component with the product name', () => {
    const wrapper = renderer.create(
        <ProductCard product={product1}>
            {
                () => (
                    <ProductTitle />
                )
            }
        </ProductCard>
      );
      expect(wrapper.toJSON()).toMatchSnapshot();
  })
  
});
