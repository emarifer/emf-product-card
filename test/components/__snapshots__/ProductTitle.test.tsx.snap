// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle it should display the component correctly with the custom title 1`] = `
<span
  className="productDescription custom3-class"
>
  Custom Product - Title
</span>
`;

exports[`ProductTitle it should show the component with the product name 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Coffee Mug - Without Image
  </span>
</div>
`;
